app/resources/v1/aliquot_resource.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# JSON-API resources wont resolve across namespaces spaces
+# so we need the Aliquot resource in the PacBio namespace
+module V1
+  # AliquotResource
+  class AliquotResource < JSONAPI::Resource
+    model_name 'Aliquot'
+    attributes :aliquot_type, :source_id, :source_type, :used_by_id, :used_by_type, :state,
+               :volume, :concentration, :insert_size, :template_prep_kit_box_barcode, :tag_id
+
+    has_one :source, polymorphic: true,
+                     polymorphic_types: %w[pacbio/requests pacbio/pools pacbio/libraries]
+    has_one :used_by, polymorphic: true,
+                      polymorphic_types: %w[pacbio/requests pacbio/pools pacbio/libraries]
+
+    has_one :tag, always_include_optional_linkage_data: true
+
+    # Aliquot polymorphism support
+    # This fixes the polymorphic relationships in json-api resources
+    # as json-api resources underscores and pluralizes the type
+    def self.resource_klass_for(type)
+      type = type.underscore.pluralize.gsub('pacbio/', '')
```

Comment:
This is really good but the main thing is that we are specifying pacbio in the top level aliquot resource which reduces flexibility. I think I might have a look at how we can use metaprogramming or composition so we can put something in aliquotable.

The other thing I am not sure of is why you are removing pacbio from the type?

Reply:
Happy for you to have a look into it.

Yeah its slightly odd but in reality we are calling this from the `v1/pacbio/aliquots` resource and in there 'requests' 'libraries' and 'pools' resolve to their local namespace resource. So the pacbio prefix isnt needed and prevents that pacbio::pacbio bug. We might be able to move the pacbio references to the `v1/pacbio/aliquots` resource if you think thats better.